Transactions/TransactionService.cs

```diff
@@ -28,57 +29,70 @@ public static TransactionService GetTransactionService()
             return _singleton;
         }
 
-        public void InitListByStringArray(string[] data)
+        /*public void InitListByStringArray(string[] data)
         {
             if (data == null) return;
            
             for (int x = 0; x < data.Length / 4; x++)
             {
-                Transaction transaction = new Transaction(data[4 * x], data[x * 4 + 1], data[x * 4 + 2]); //transactionType, amount, transactionTitle
+                Transaction transaction = new Transaction(data[4 * x], data[x * 4 + 1], data[x * 4 + 2], data[x * 4 + 3]); //transactionType, amount, transactionTitle
                 list.Add(transaction);
                 //Console.WriteLine(transaction.Title);
                 //Console.WriteLine(data[4 * x] + x);
             }
-        }
+        }*/
 
         public List<Transaction> GetListWithTitleFiltered(string filter)
         {
             var query = list.Where(oh => oh.Title.ToLower().Contains(filter)).ToList();
-            Console.WriteLine(query.Count);
+            Console.WriteLine("Query count:" + query.Count);
             return query;
 
         }
 
         public void LoadTransactionsListFromTextFile()
         {
+            list.Clear();
             ListItemTransactions[] listItems = new ListItemTransactions[20];
             TextFileReader reader = new TextFileReader();
             string[] data = reader.FetchStringArrayByLocation(System.AppDomain.CurrentDomain.BaseDirectory + @"..\..\Data\Transactions.txt");
-
-            this.InitListByStringArray(data);
-
+            if (data == null) return;
+            for (int x = 0; x < data.Length; x++)
+            {
+                Transaction transaction = JsonConvert.DeserializeObject<Transaction>(data[x]);
+                if(transaction != null)
+                {
+                    list.Add(transaction);
+                }
+                
+            }
         }
         public List<Transaction> GetTransactionsList()
         {
-            return list;
+            return list == null ? null : list;
+            //return list ?? null;
         }
 
-        public void AddNewTransaction(string transactionType, string transactionName, string transactionAmount)
+        public void AddNewTransaction(string transactionType, string transactionName, string transactionAmount, string category)
         {
-            StreamWriter w = File.AppendText(System.AppDomain.CurrentDomain.BaseDirectory + @"..\..\Data\Transactions.txt");
-            if (transactionType   != "" &&
-                transactionName   != "" &&
-                transactionAmount != "")
+            
+            if (transactionType != "" && transactionName != "" && transactionAmount != "")
```

Comment:
what if the properties are null, or contains some whitespace? Consider using` string.IsNullOrWhiteSpace `for such checks